src/Transport/GrpcTransport.php

```diff
@@ -159,6 +159,8 @@ public static function build(string $apiEndpoint, array $config = [])
      */
     public function startBidiStreamingCall(Call $call, array $options)
     {
+        $this->verifyUniverseDomain($options);
```

Comment:
Clarification: Does this need to happen for every call because credentials can come through the call options? Or is there a singular place this could be verified and not on every call?

Reply:
Good question - this was a specific choice because I wanted to avoid adding a network call for the credentials to get the network domain before a call was made - to avoid adding latency where it previously didn't exist. 

As a result, the universe domain is checked at call time. But it's only checked once per request (see [CredentialsWrapper::checkUniverseDomain](https://github.com/googleapis/gax-php/blob/dfca60f46fed8d9a4d9d2ece7286f18d53d3ebad/src/CredentialsWrapper.php#L277) and never again.

Reply:
awesome, sgtm, good choice!